File: 1.1/src/cz/mg/vulkan/PFNvkQueueSubmit.java
package cz.mg.vulkan;

public class PFNvkQueueSubmit extends VkFunctionPointer {
    public PFNvkQueueSubmit() {
    }

    public PFNvkQueueSubmit(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkQueueSubmit(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkQueueSubmit(long value) {
        setValue(value);
    }

    public PFNvkQueueSubmit(VkInstance instance) {
        super(instance, new VkString("vkQueueSubmit"));
    }

    public void call(VkQueue queue, VkUInt32 submitCount, VkSubmitInfo pSubmits, VkFence fence, VkResult rval){
        call(getValue(), queue != null ? queue.getVkAddress() : VkPointer.getNullAddress(), submitCount != null ? submitCount.getVkAddress() : VkPointer.getNullAddress(), pSubmits != null ? pSubmits.getVkAddress() : VkPointer.NULL, fence != null ? fence.getVkAddress() : VkPointer.getNullAddress(), rval != null ? rval.getVkAddress() : VkPointer.getSinkAddress());
    }

    protected static native void call(long vkaddress, long queue, long submitCount, long pSubmits, long fence, long rval);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkQueueSubmit.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkQueueSubmit_callNative(JNIEnv* env, jclass jc, jlong address, jlong queue, jint submitCount, jlong pSubmits, jlong fence){
    (void)env;
    (void)jc;
    PFN_vkQueueSubmit f = (PFN_vkQueueSubmit)jniLongToFunctionPointer(address);
    return f(
        *((VkQueue*)jniLongToPointer(queue)),
        (uint32_t)submitCount,
        ((VkSubmitInfo*)jniLongToPointer(pSubmits)),
        *((VkFence*)jniLongToPointer(fence))
    );
}

